core/src/main/java/org/apache/accumulo/core/data/ArrayByteSequence.java

```diff
@@ -168,10 +161,7 @@ public void reset(byte[] data, int offset, int length) {
   @Override
   public ByteSequence subSequence(int start, int end) {
 
-    if (start > end || start < 0 || end > length) {
-      throw new IllegalArgumentException("Bad start and/end start = " + start + " end=" + end
-          + " offset=" + offset + " length=" + length);
-    }
```

Comment:
Is `length` correct here, or should it changed to `data.length`? Wasn't completely sure.

Reply:
You commented on the wrong line in GitHub, so it's not showing me the line you're asking about. If you want to comment on a range of lines, you can click on the first line and drag down until the last line, and it will show the whole chunk. To your question, I think you did it correctly. `length` refers to the length of the limited "view" of the backing array, and the start and end of the subsequence are coming from that "view" of the backing array, not the full backing array, and have to fit in that view's length.